5-Access-Authentication-and-Authorization/5.4-Configure-PAM.bats

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bats
+
+@test "5.4.1 Ensure password creation requirements are configured (Automated)" {
+    local MINLENGTH=$(grep '^\s*minlen\s*' /etc/security/pwquality.conf)
+    MINLENGTH=(${MINLENGTH//minlen = / }) # get the number from the string
+    local PWQUALITY=$(grep '^\s*minclass\s*' /etc/security/pwquality.conf)
+    PWQUALITY=(${PWQUALITY//minclass = / }) # get the number from the string
+    if [[ "$MINLENGTH" -lt 14 ]] || [[ "$PWQUALITY" -lt 4 ]]; then
+        local CREDITPWQUALITY=$(grep -E '^\s*[duol]credit\s*' /etc/security/pwquality.conf)
+        [[ "$CREDITPWQUALITY" != "" ]]
+        [[ "$CREDITPWQUALITY" == *"dcredit = -1"* ]]
+        [[ "$CREDITPWQUALITY" == *"ucredit = -1"* ]]
+        [[ "$CREDITPWQUALITY" == *"lcredit = -1"* ]]
+        [[ "$CREDITPWQUALITY" == *"ocredit = -1"* ]]
+        local RETRIES=$(grep -E '^\s*password\s+(requisite|required)\s+pam_pwquality\.so\s+(\S+\s+)*retry=[1-3]\s*(\s+\S+\s*)*(\s+#.*)?$' /etc/pam.d/common-password)
+        [[ "$RETRIES" == "password"*"requisite"*"retry="[0-3] ]]
```

Comment:
The `[0-3]` part looks like regex to me. If that's true, don't we need `"$RETRIES" =~` here?

Reply:
It works as well with `==`.

Reply:
But why?